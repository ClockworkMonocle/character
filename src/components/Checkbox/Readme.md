A simple style wrapper around `<input type="checkbox" />`. Can also use with a `label` via props.


Basic checkbox:

```js
<Checkbox />
```

Checkbox with label (default placement):

```js
<Checkbox label="You should select me!" />
```

Checkbox as controlled component:

```js
initialState = { checked: true };
const toggle = () => setState({ checked: !state.checked });

<Checkbox checked={state.checked} onChange={toggle} />
```
