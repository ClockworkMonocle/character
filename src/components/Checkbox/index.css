
.checkbox {
  cursor: pointer;
  position: relative;
  line-height: normal;
  display: flex;
  align-items: center;
}

.checkbox > input[type="checkbox"] {
  display: none;
}

.checkbox_label {
  font-size: 0.80em;
  margin-left: 8px;
  font-weight: bold;
}

.checkbox_indicator {
  display: block;
  width: 1em;
  height: 1em;
  border: 1px solid var(--black);
  content: '';
  cursor: pointer;
  user-select: none;
  transition: background-color, border-color 300ms ease;
}

.checkbox > input[type="checkbox"]:checked ~ .checkbox_indicator {
  background-image: url('data:image/svg+xml;charset=utf-8, %3Csvg%20width%3D%2210%22%20height%3D%229%22%20viewBox%3D%220%200%2010%209%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M1%204l3%203%205-6%22%20stroke-width%3D%222%22%20stroke%3D%22%23FFF%22%20fill%3D%22none%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E');
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 10px;
  vertical-align: middle;
  background-color: var(--attack-color);
  border-color: var(--attack-color);
}
