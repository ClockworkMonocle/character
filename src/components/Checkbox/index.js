import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

export default function Checkbox({ label, id, ...props }) {
  return (
    <label className={cn('checkbox', label && 'checkbox--hasLabel')} htmlFor={id}>
      <input
        id={id}
        aria-checked="false"
        tabIndex="0"
        type="checkbox" {...props}
      />
      <span className="checkbox_indicator" />
      {label && <span className="checkbox_label">{label}</span>}
    </label>
  );
}

Checkbox.propTypes = {
  label: PropTypes.string,
};
