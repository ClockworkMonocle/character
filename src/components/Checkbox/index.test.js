import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './index';

test('<Checkbox /> renders without label', () => {
  expect(shallow(<Checkbox />)).toMatchSnapshot();
});

test('<Checkbox /> renders with label', () => {
  expect(shallow(<Checkbox label="Heyo" />)).toMatchSnapshot();
});
