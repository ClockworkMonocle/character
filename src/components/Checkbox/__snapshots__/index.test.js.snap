// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders with label 1`] = `
<label
  className="checkbox checkbox--hasLabel"
>
  <input
    aria-checked="false"
    tabIndex="0"
    type="checkbox"
  />
  <span
    className="checkbox_indicator"
  />
  <span
    className="checkbox_label"
  >
    Heyo
  </span>
</label>
`;

exports[`<Checkbox /> renders without label 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked="false"
    tabIndex="0"
    type="checkbox"
  />
  <span
    className="checkbox_indicator"
  />
</label>
`;
